### ISSUE TITLE
Set std::terminate to Parallel::abort?

### ISSUE THREAD

**Issue by nilsdeppe** [sxs-collaboration/spectre#195]
We should look at setting general termination to use Charm++ abort so things work correctly in parallel.

**Comment by kidder** [comment]
This is done in RunTests.cpp and should be done in all executables in an initnode function along with enabling FPEs
```  cpp
std::set_terminate(
    []() { Parallel::abort("Called terminate. Aborting..."); });
enable_floating_point_exceptions();
```

**Comment by nilsdeppe** [comment]
This has been reported upstream with some discussion: https://charm.cs.illinois.edu/redmine/issues/1681

**Comment by nilsdeppe** [comment]
What's left on this issue is:

- [x] Check that initnode is the right function to set terminate and FPEs in using both SMP and non-SMP builds on Wheeler using two nodes and having each test trigger a terminate or FPE on a single core, going through all cores one at a time.

**Comment by nilsdeppe** [comment]
@kidder and I figured this out. Fixed in #541
